src/payment/instrument/instrument-selector.ts

```diff
@@ -1,40 +1,75 @@
+import { createSelector } from '../../common/selector';
+import { memoizeOne } from '../../common/utility';
+
 import Instrument from './instrument';
-import InstrumentState, { InstrumentMeta } from './instrument-state';
+import InstrumentState, { DEFAULT_STATE, InstrumentMeta } from './instrument-state';
+
+export default interface InstrumentSelector {
+    getInstruments(): Instrument[] | undefined;
+    getInstrumentsMeta(): InstrumentMeta | undefined;
+    getLoadError(): Error | undefined;
+    getDeleteError(instrumentId?: string): Error | undefined;
+    isLoading(): boolean ;
+    isDeleting(instrumentId?: string): boolean;
+}
+
+export type InstrumentSelectorFactory = (state: InstrumentState) => InstrumentSelector;
 
-export default class InstrumentSelector {
-    constructor(
-        private _instruments: InstrumentState
-    ) {}
+export function createInstrumentSelectorFactory(): InstrumentSelectorFactory {
+    const getInstruments = createSelector(
+        (state: InstrumentState) => state.data,
+        instruments => () => instruments
+    );
 
-    getInstruments(): Instrument[] | undefined {
-        return this._instruments.data;
-    }
+    const getInstrumentsMeta = createSelector(
+        (state: InstrumentState) => state.meta,
+        meta => () => meta
+    );
 
-    getInstrumentsMeta(): InstrumentMeta | undefined {
-        return this._instruments.meta;
-    }
+    const getLoadError = createSelector(
+        (state: InstrumentState) => state.errors.loadError,
+        loadError => () => loadError
+    );
 
-    getLoadError(): Error | undefined {
-        return this._instruments.errors && this._instruments.errors.loadError;
-    }
+    const getDeleteError = createSelector(
+        (state: InstrumentState) => state.errors.failedInstrument,
```

Comment:
unrelated to this PR, but maybe we can rename for consistency?  `deleteInstrumentId`

Reply:
Nice pickup. I'll address it separately :)